### PR TITLE
MatrixTable column operations all go through select

### PR DESCRIPTION
**Pull request by catoverdrive**
rewrite select/annotate/drop to go through `_select_cols` in Python

### CODE REVIEW

File: src/main/scala/is/hail/variant/MatrixTable.scala [hunk 1]
```diff
@@ -799,43 +799,6 @@ class MatrixTable(val hc: HailContext, val ast: MatrixIR) {
...
-        }
-      i += 1
-    }
-
-    val saFields = finalType.fieldNames.toSet
-    copyMT(colValues = colValues.copy(value = newAnnotations, t = TArray(finalType)),
-      matrixType = matrixType.copy(colKey = colKey.filter(saFields.contains), colType = finalType))
-  }
-
   def annotateCols(annotations: Map[Annotation, Annotation], signature: Type, root: String): MatrixTable = {
```
Comment (jigold) [inline src/main/scala/is/hail/variant/MatrixTable.scala]: Should this get added to RichMatrixTable? Is it still used anywhere?

Comment (catoverdrive) [reply src/main/scala/is/hail/variant/MatrixTable.scala]: I did. (slightly different form, but I updated the tests accordingly).

Comment (jigold) [reply src/main/scala/is/hail/variant/MatrixTable.scala]: Can you delete it from MatrixTable.scala?

Comment (jigold) [reply src/main/scala/is/hail/variant/MatrixTable.scala]: I mean `annotateCols`

Comment (catoverdrive) [reply src/main/scala/is/hail/variant/MatrixTable.scala]: oh, this one. my bad I thought you were referring to the other one. I'll move this also

Comment (catoverdrive) [reply src/main/scala/is/hail/variant/MatrixTable.scala]: (done)

**Pull request merged**